cmd/cored/main.go

```diff
@@ -152,9 +152,26 @@ func main() {
 		internalDN = &x509Cert.Subject
 	}
 
+	// TODO(kr): make core.UseTLS take just an http client
+	// and use this object in it.
+	httpClient := new(http.Client)
+	httpClient.Transport = &http.Transport{
+		TLSClientConfig: tlsConfig,
+
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		MaxIdleConns:          100,
```

Comment:
Yeah we prob should be more deliberate about them at some point. I copied these from https://godoc.org/net/http#DefaultTransport which is what we were using before, to avoid any unintended changes in behavior.